sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java

```diff
@@ -97,39 +111,73 @@ private boolean isForSigning(Key key) {
     return StringUtils.isBlank(key.getUse()) || "sig".equals(key.getUse());
   }
 
-  private boolean isRsaKeyType(Key key) {
-    return "RSA".equals(key.getKty());
+  private boolean isSupportedKeyType(Key key) {
+    return SUPPORTED_KTY.contains(key.getKty());
+  }
+
+  private boolean isSupportedAlg(Key key) {
+    return SUPPORTED_ALG.contains(key.getAlg());
   }
 
-  private boolean isRsa256Key(Key key) {
-    // We only support RSA256, if blank we assume it to be RS256.
-    return StringUtils.isBlank(key.getAlg()) || "RS256".equals(key.getAlg());
```

Comment:
The breaking change is actually here. So far we assume if the `"alg": "" ` is provided in the JWK, the RSA256 is used as default. Which is in my opinion a bad behavior. The "alg" should be "none" (also not supported so far) or have an existing alg. 

We would throw an exception if none of the supported algs couldnt be determined.

Reply:
We could still fallback to `RSA256`, but this feels not right in my opinion. The parameter is OPTIONAL in the specification: https://datatracker.ietf.org/doc/html/rfc7517#section-4.4, which means for me that we are using a not signed JWT so not a JWS here. Which is the equivalent of "none". Which could be an proper default value then. 

See the list of algs here: https://www.iana.org/assignments/jose/jose.xhtml

Reply:
For security reasons we should not support `alg: "none"`.

Reply:
We do fix a non-compliant implementation by this "breaking change", don't we? We could argue that this is not a breaking change then. What do you think?